verilog/analysis/checkers/parameter_name_style_rule.cc

```diff
@@ -40,27 +38,47 @@
 namespace verilog {
 namespace analysis {
 
+VERILOG_REGISTER_LINT_RULE(ParameterNameStyleRule);
+
 using verible::LintRuleStatus;
 using verible::LintViolation;
 using verible::SyntaxTreeContext;
 using Matcher = verible::matcher::Matcher;
 
-// Register ParameterNameStyleRule.
-VERILOG_REGISTER_LINT_RULE(ParameterNameStyleRule);
+// PascalCase, may end in _[0-9]+
+static constexpr absl::string_view localparam_default_regex =
```

Comment:
Note, that all constants are always named k-prefix-with-camel-case throughout this project `kFooBar`; so this would be `kLocalparamDefaultRegex` (I think I have overlooked that in the other reviews, it should be corrected there as well).

I think we should have an indirection here with naming that reflects the meaning of the regexes and then the meaning of the default value.
So

```c++
static constexpr absl::string_view kCamelCaseRegex = ...;
static constexpr absl::string_view kUpperSnakeCaseRegex = ...;

static constexpr absl::string_view kLocalparamDefaultRegex = kCamelCaseRegex;
static constexpr absl::string_view kParameterDefaultRegex = kUpperSnakeCaseRegex;
```

Not only makes it more readable here what we mean, but also in the Configuration() below, if we use the named styles (see comment there), it is much easier to refer to it accordingly.

Reply:
No worries, I'll fix that up in the open PR's. Would you prefer a single or multiple follow up PR's to fix the others that sneaked through?

Reply:
One change that touches all is fine.

Reply:
No problem, see PR #2236